pkgs/by-name/op/openmpi/package.nix

```diff
@@ -152,14 +152,17 @@ stdenv.mkDerivation (finalAttrs: {
       # The file names we need to iterate are a combination of ${p}${s}, and there
       # are 7x3 such options. We use lib.mapCartesianProduct to iterate them all.
       fileNamesToIterate = {
-        p = [
-          "mpi"
-          "shmem"
-          "osh"
-        ];
-        s =
+        p =
           [
-            "CC"
+            "mpi"
+          ]
+          ++ lib.optionals stdenv.hostPlatform.isLinux [
+            "shmem"
+            "osh"
+          ];
+        s =
+          lib.optional stdenv.hostPlatform.isLinux "CC"
```

Comment:
Nit: Could you use `lib.optionals`? It should also be possible to put it after the `lib.optionals fortranSupport` and the formatting + diff will be more consistent.

Reply:
This review comment wasn't addressed.